orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/AccountIntervention.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.di.orchestration.shared.entity;
+
+public class AccountIntervention {
+
+    private final AccountInterventionDetails details;
+    private final AccountInterventionState state;
+    private final AccountInterventionStatus status;
+    private final boolean ignoreResetPassword;
+
+    public AccountIntervention(AccountInterventionState state) {
+        this.details =
+                new AccountInterventionDetails(
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE,
+                        "",
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE);
+        this.state = state;
+        this.ignoreResetPassword = false;
+        this.status = deduceStatus();
+    }
+
+    public AccountIntervention(
+            AccountInterventionDetails details,
+            AccountInterventionState state,
+            Long passwordResetTime) {
+        this.details = details;
+        this.state = state;
+        this.ignoreResetPassword = details.appliedAt() < passwordResetTime;
+        this.status = deduceStatus();
+    }
+
+    public AccountIntervention(AccountInterventionState state, Long passwordResetTime) {
+        this.details =
+                new AccountInterventionDetails(
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE,
+                        "",
+                        Long.MIN_VALUE,
+                        Long.MIN_VALUE);
+        this.state = state;
+        this.ignoreResetPassword = details.appliedAt() < passwordResetTime;
+        this.status = deduceStatus();
+    }
```

Comment:
This constructor feels a little odd. Given it's only used in one test I think I'd construct the details object in the test and remove this constructor.